Sources/StoryboardInstantiatable.swift

```diff
@@ -73,6 +73,36 @@ extension DependencyContainer {
   public func resolveDependencies<T>(of instance: T, tag: Tag? = nil) throws {
     _ = try resolve(tag: tag) { (_: () throws -> T) in instance }
   }
+  
+  /**
+   Register factory for type `T` which has to conform to `StoryboardInstantiatable` and associate it with an optional tag.
+   
+   - parameters:
+   - scope: The scope to use for instance created by the factory. Default value is `Shared`.
+   - type: Type to register definition for. Default value is return value of factory.
+   - tag: The arbitrary tag to associate this factory with. Pass `nil` to associate with any tag. Default value is `nil`.
+   - factory: The factory that produces instance of `type`. Will be used to resolve instances of `type`.
+   
+   - returns: A registered definition.
+   
+   - note: You should cast the factory return type to the protocol you want to register it for
+   (unless you want to register concrete type) or provide `type` parameter.
+   
+   - seealso: `Definition`, `ComponentScope`, `DependencyTagConvertible`
+   
+   **Example**:
+   ```swift
+   //Register MyViewController
+   container.register(tag: "myVC") { MyViewController() }
+   .resolvingProperties { container, controller in
+      controller.service = try container.resolve() as MyViewControllerService
+   }
+   
+   ```
+   */
+  public func registerViewController<T: StoryboardInstantiatable>(_ scope: ComponentScope = .shared, type: T.Type = T.self, tag: DependencyTagConvertible? = nil, factory: @escaping () throws -> T) -> Dip.Definition<T, ()> {
```

Comment:
Few comments regarding this implementation:
- scope does not really matter in this case because container will always use instance created from a storyboard
- factory is also not needed as type must be NSObject in this context and we can use it's `init` internally just to satisfy register method parameters (it will never be called any way)
- not only view controllers can be injected in this way (though they are a primary use case), so I would name this method smth like `register(storyboardType:)`

So it will be something like this:

```swift
  public func register<T: NSObject>(storyboardType type: T.Type, tag: DependencyTagConvertible? = nil) -> Definition<T, ()> where T: StoryboardInstantiatable {
    return register(.unique, type: type, tag: tag, factory: { T() })
  }
```